pkg/cmd/server/api/types.go

```diff
@@ -699,3 +699,93 @@ type AssetExtensionsConfig struct {
 	// Web Console's context root. Defaults to false.
 	HTML5Mode bool
 }
+
+type LDAPSyncConfig struct {
+	api.TypeMeta
+	// LDAPClientInfo contains the information about connection to the LDAP server.
+	// The contained URL will be parsed for scheme, host and port. The syntax of the URL is:
+	// `scheme://host:port/`, where `scheme` is either `ldap` or `ldaps`. Any other encoding in the
+	// URL will be ignored.
+	LDAPClientInfo LDAPPasswordIdentityProvider
+
+	// GroupNameAttributes (optionally) maps LDAP group entries to OpenShift Groups
+	GroupNameAttributes []string
```

Comment:
> Should it remain as an optional field here or should there be another EmbeddedObject to hold other schema-specific config? Should there be just one catch-all schema-specific config object?

Schema specific config options should be contained in the schema specific config objects.  I don't think having a congruent name is worth having sometimes valid config options.  I also don't have an issue with including this in an object called `Schema1QueryTemplates`, so I don't think you lose your nice name.